### ISSUE TITLE
some questions about NHV vocoder

### ISSUE THREAD

**Issue by LqNoob** [zjlww/zjlww.github.io#2]
Hi,  
After reading your article and the online supplement [https://zjlww.github.io/is2020/](https://zjlww.github.io/is2020/), I have met some quetions as follows:
1. In part2.2, "The neural network only predicts low-quefrency coefficients. The high-quefrency cepstrum coefficients are set to zero.", what's this message mean? How to set high-quefrency cepstrum coefficients to zero？
2. In computational complexity page,  when calculating ‘nhv_filters’,  what does it mean to add a value of 2048 and what calculation is it derived from?
3. In training details page, what does ‘weighted by 4.0’ mean  in line ‘Weight of 𝐿𝑅 and 𝐿𝐺, 𝐿𝐷’？

Thanks

**Comment by zjlww** [comment]
Sorry for the confusion.
For the first problem, [the last page in this slide](https://slideplayer.com/slide/13590881/) shows a complex cepstrum analyzed from a segment of speech. The high-quefrency coefficients refer to those corresponding to the high absolute value of quefrency. In our work:
    1. The neural network only predicts low-quefrency coefficients (since higher quefrency corresponds to unnecessary details in the impulse response).
    2. And then pad zeros on both sides of the predicted short complex cepstrum (nFFT long).
    3. Then the padded long complex cepstrum is inverted back to impulse responses (nFFT long).
    4. The impulse responses are used to filter the excitation signals.

**Comment by zjlww** [comment]
For the second problem, if you mean these

  > ln[24]: nhv_filters += (complex_multiply(2048) + fft(2048) + **_2048_**) * 2 / 128
  > ln[25]: nhv_filters += (fft(2048) + complex_multiply(2048) + fft(2048) + **_2048_**) / 1024

2048, the reasons are the same. They both come from the overlap and add operation after the FFT.

**Comment by zjlww** [comment]
For the third problem, L_R is not weighted, L_G and L_D are weighted by 4. So the full loss function would be L_R + 4.0 (L_G + L_D).

**Comment by LqNoob** [comment]
@zjlww 
Thank you for your answer and kind of work, this is very helpful for me to implement your paper "Neural Homomorphic Vocoder".
